src/bootstrap_handler.rs

```diff
@@ -129,8 +121,27 @@ impl BootstrapHandler {
 
     fn write_bootstrap_file(&mut self, mut contacts: Contacts) -> io::Result<()> {
         contacts = contacts.clone().into_iter().unique().collect();
-        self.file_path = BootstrapHandler::get_file_path();
-        let mut file = try!(File::create(&self.file_path));
+        let _ =  fs::create_dir_all(self.file_path.parent().unwrap());
+        let mut file = match File::create(&self.file_path) {
+            Ok(created_file) => created_file,
+            Err(e) => {
+                let user_dir = path::PathBuf::from(utils::user_app_dir().unwrap()
+                                    .join(self.file_path.file_name().unwrap()));
+                if self.file_path != user_dir {
+                    match fs::create_dir_all(user_dir.parent().unwrap()) {
+                        Ok(_) => {
+                            let file =  try!(File::create(&user_dir));
+                            self.file_path = user_dir;
+                            file
+                        },
+                        Err(e) => return Err(e)
```

Comment:
Since we're just returning the same error as is generated by `create_dir_all`, shouldn't we just be doing a `try!(create_dir_all(...))`?